common/src/p2p/peer.rs

```diff
@@ -40,15 +40,19 @@ pub trait Store {
 
     /// Seed the peer store with addresses.
     /// Fails if *none* of the seeds could be resolved to addresses.
-    fn seed<S: net::ToSocketAddrs>(
+    fn seed<S: net::ToSocketAddrs + std::fmt::Debug>(
         &mut self,
-        seeds: impl Iterator<Item = S>,
+        mut seeds: impl Iterator<Item = S>,
         source: Source,
     ) -> io::Result<()> {
-        let mut error = None;
+        let mut fail = false;
         let mut success = false;
-
-        for seed in seeds {
+        // TODO: Introduce a step that mean how many combination
+        // of services flags we need to try before check the seeds without
+        // services flags.
+        // for the moment it is 2, but we need a more general solution.
+        while let Some(seed) = seeds.next() {
+            debug!("Resolving DNS seed {:?}", seed);
```

Comment:
I think this PR it is not completed, because I'm missing the services flag that it is Supported by the network

Here there is still the services flag none, and this it wrong I think

```
match seed.to_socket_addrs() {
                Ok(addrs) => {
                    success = true;
                    for addr in addrs {
                        self.insert(
                            addr.ip(),
                            KnownAddress::new(
                                Address::new(&addr, ServiceFlags::NONE),
                                source,
                                None,
                            ),
                        );
                    }
```

Maybe this solution proposed is totally wrong because I'm missing the services flag in the address. Mh!